backend/src/load.rs

```diff
@@ -0,0 +1,404 @@
+use std::collections::{HashMap, HashSet};
+use std::cmp::{Ordering, max};
+use std::io::Read;
+
+use chrono::{Duration, NaiveDateTime};
+use json::{self, JsonValue};
+
+use util::start_idx;
+use errors::*;
+
+const WEEKS_IN_SUMMARY: usize = 12;
+
+/// Contains the data loaded from the file system.
+/// Useful to avoid passing around each individual field, instead passing the
+/// entire struct.
+pub struct ServerData {
```

Comment:
Sounds good. I'll go with `InputData` since we do processing before it goes in here (summary construction, parsing the JSON, etc.).